stk/calculators/optimization/optimizers.py

```diff
@@ -662,88 +672,457 @@ class GFNXTB(Optimizer):
+        else:
+            solvent = ''
+
+        cmd = (
+            f'{memory} {self.xtb_path} {xyz} --gfn {self.gfn_version} '
+            f'{optimization} --parallel {self.num_cores} '
+            f'--etemp {self.electronic_temperature} '
+            f'{solvent} --chrg {self.charge} '
+            f'--uhf {self.num_unpaired_electrons}'
+        )
+
+        with open(out_file, 'w') as f:
+            # Note that sp.call will hold the program until completion
+            # of the calculation.
+            sp.call(
+                cmd,
+                stdin=sp.PIPE,
+                stdout=f,
+                stderr=sp.PIPE,
+                # Uses the shell if unlimited_memory is True to be run
+                # multiple commpands in one subprocess.
+                shell=self.unlimited_memory
+            )
+
+    def _run_optimizations(self, mol, conformer):
+        """
+        Run loop of optimizations of the molecule `mol` using xTB.
+
+        Parameters
+        ----------
+        mol : :class:`.Molecule`
+            The molecule to be optimized.
+
+        conformer : :class:`int`
+            The conformer to use.
+
+        Returns
+        -------
+        None : :class:`NoneType`
+
+        """
+        runs = self.max_runs if self.max_runs is not None else 0
+        for run in range(runs):
+            xyz = f'input_structure_{run+1}.xyz'
+            out_file = f'optimization_{run+1}.output'
+            mol.write(xyz, conformer=conformer)
+            self._run_xtb(xyz=xyz, out_file=out_file)
+            # Check if the optimization is complete.
+            coord_file = 'xtbhess.coord'
+            coord_exists = os.path.exists(coord_file)
+            output_xyz = 'xtbopt.xyz'
+            opt_incomplete = self._incomplete(out_file)
+            if opt_incomplete and coord_exists:
+                # The calculation is incomplete.
+                # Update mol from xtbhess.coord and continue.
+                mol.update_from_turbomole(
+                    path=coord_file,
+                    conformer=conformer
+                )
+            else:
+                # Update mol from xtbopt.xyz.
+                mol.update_from_xyz(
+                    path=output_xyz,
+                    conformer=conformer
+                )
+                if opt_incomplete:
+                    # If the negative frequencies are small, then GFN
+                    # may not produce the restart file. If that is the
+                    # case, exit optimization loop and warn.
+                    self.incomplete.add((mol, conformer))
+                    logging.warning(
+                        'Small negative frequencies present in '
+                        f'{mol.name} conformer {conformer}.'
+                    )
+                    return False
+                else:
+                    # Calculation is complete.
+                    return True
+        return None
```

Comment:
Don't write return None, delete this line entirely, it is completely redundant

Not writing it  does the exact same thing

even 

```
return 
```

followeding by nothing is the same as return None.